[Workers] Updated KV section to address reading a deleted key (#8603)
cloudflare/cloudflare-docs: content/workers/runtime-apis/kv.md

This method returns a promise that you should `await` on in order to verify successful deletion.

> Note that attempting to read a deleted key from KV will result in a `404 Error`.

Reviewer: Attempt to read a deleted key from KV will result in a `404 Error`

> Note that attempting to read a deleted key from KV will result in a `404 Error`.

Reviewer 2: ```suggestion
Attempting to read a deleted key from KV will result in a `404 Error`.
```

> Note that attempting to read a deleted key from KV will result in a `404 Error`.

Reviewer 2: ```suggestion
Attempting to read a deleted key from KV will result in a `404 Not Found` error.
```

Removing 'Note that' as it is largely unnecessary. If you wanted to highlight this point, I'd recommend putting this into an aside.

> You can also [delete key-value pairs from the command line with Wrangler](/workers/wrangler/workers-kv/) or [via the API](/api/operations/workers-kv-namespace-delete-key-value-pair).

Reviewer 2: ```suggestion
You can [delete key-value pairs from the command line with Wrangler](/workers/wrangler/workers-kv/) or [via the API](/api/operations/workers-kv-namespace-delete-key-value-pair).
```
